packages/node-sdk/src/client.ts

```diff
@@ -395,13 +395,8 @@ export class BucketClient {
    *
    * @param context - The context to update.
    */
-  private async syncContext({
-    enableTracking = true,
-    ...context
-  }: ContextWithTracking) {
-    checkContextWithTracking({ enableTracking, ...context });
-
-    if (!enableTracking) {
+  private async syncContext(options: ContextWithTracking) {
+    if (!options.enableTracking) {
```

Comment:
Assuming here that the enable tracking is either true or false and cannot be undefined.

In these cases I love them `assert`s to make this clear to the reader of the code and also find bugs.

Reply:
OK, adding `checkContextWithTracking` to be sure.